scripts/google/covid_mobility/covidmobility.py

```diff
@@ -0,0 +1,235 @@
+
+        # If no dcid, skip the row.
+        if not region_dcid:
+            continue
+
+        # If no date, skip the row.
+        if not date:
+            continue
+
+        # Iterate through all places in the row.
+        for place in PLACE_CATEGORIES:
+            # Type of place using Schema notation.
+            schema_place = PLACE_CATEGORIES[place]
+
+            population_id = f"{region_dcid}_{schema_place}"
+            population_id = convert_to_ascii(population_id)
+
+            observation_id = f"{population_id}_{date}"
+
+            # If this is the first time vieweing this dcid.
+            # Write the population node for the place.
+            if region_dcid not in visited_dcids:
+                f_output.write(
+                    f"Node: {population_id}\n"
+                    "typeOf: schema:StatisticalPopulation\n"
+                    f"location: dcid:{region_dcid}\n"
+                    "populationType: dcs:PlaceVisitEvent\n"
+                    f"placeCategory: dcs:{schema_place}\n\n")
+
+            # Get the value for the current place.
+            value = row[place]
+
+            # If the value is None, skip the row.
+            if not value:
+                continue
+
+            # Write observation node for value.
+            f_output.write(
+                f"Node: {observation_id}\n"
+                "typeOf: schema:Observation\n"
+                f"observedNode: l:{population_id}\n"
+                f'observationDate: "{date}"\n'
+                "measuredProperty: dcs:covid19MobilityTrend\n"
+                f"measuredValue: {value}\n"
+                "unit: dcs:Percent\n\n")
+
+        # Add dcid to the list of visited.
+        visited_dcids.add(region_dcid)
+
+    f_input.close()
+    f_output.close()
+
+
+def download_data(url: str, download_as: str) -> None:
+    """Download the data file from the input url.
+
+    Args:
+        url (str): URL of the file.
+        download_as (str): path to save the file.
+    """
+
+    # Request data from url.
+    data: str = urlopen(url).read().decode('utf-8')
+
+    # Store the data as a download_as.
+    with open(download_as, 'w+') as input_file:
+        input_file.write(data)
+
+
+def get_region_dcid(sub_region_2: str,
+                    sub_region_1: str,
+                    country_code: str) -> str:
+    """Returns the dcid for the region.
+
+
+    Args:
+        sub_region_2 (str): Usually a US County.
+        sub_region_1 (str): Usually a US State or a country's province.
+        country_code (str): Country Code. Examples: ES, US.
+
+    Returns:
+        str: the dcid of the region.
+    """
+
+    # Get rid of sub_region_1 whitespaces.
+    # Hashmap keys don't contain whitespaces.
+    sub_region_1 = sub_region_1.replace(" ", "")
+
+    if sub_region_1:
+        # Only US sub-regions are allowed.
+        if country_code != 'US' or sub_region_1 not in USSTATE_TO_ALPHA2:
+            return None
```

Comment:
To confirm, does the dataset itself not have sub-regions, or our mappings don't?

Reply:
The dataset includes sub-regions for other countries, but we don't have the maps for those.

So for now we decided to only care about US subregions, although in the future we want to allow international subregions too.

Reply:
Please add this comment to the readme, with a TODO to expand to international subregions. That is going to be come important very soon.